src/shrpx.cc

```diff
@@ -386,6 +403,9 @@ void exec_binary() {
     if (pid == -1) {
       auto error = errno;
       LOG(ERROR) << "fork() failed errno=" << error;
+    } else {
+      // update PID tracking information in systemd
+      shrpx_sd_notifyf(0, "MAINPID=%s\n", util::utos(pid).c_str());
```

Comment:
Thank you for clarification.
One thing that worries me is that if new process exits because of configuration error, then systemd gets non existent PID, and still old process continues to exist.  systemd may respawn nghttpx process if it finds the managed process is gone, but because old process has still hold ports, new process cannot be started.
For this PR, it is OK for me since we use SIGHUP for reloading with systemd, and hot swapping should still be done manually.